server/matchmaker/matchmaker_queue.py

```diff
@@ -184,12 +184,17 @@ async def find_matches(self) -> None:
 
         self._register_unmatched_searches(unmatched_searches)
 
-        number_of_matches = len(matches)
-        metrics.matches.labels(self.name).set(number_of_matches)
-
         for search1, search2 in matches:
-            # TODO: Move this into algorithm, then don't need to recalculate
-            # quality_with? Probably not a major bottleneck though.
+            self._report_party_sizes(search1)
+            self._report_party_sizes(search2)
+
+            rating_imbalance = abs(search1.cumulative_rating - search2.cumulative_rating)
+            metrics.match_rating_imbalance.labels(self.name).set(rating_imbalance)
```

Comment:
Not sure if this will be that useful. Setting a Gauge in a loop like this will make it so you basically only ever see the value of the last one set. Maybe a `Histogram` would be more useful? Same for `match_rating_variety` below.

Reply:
Each metric generates a time series. https://prometheus.io/docs/concepts/data_model/
So we won't miss any data. The loop will make it so the timestamps are really close for batches of data, but this should not be a problem.